partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,365 @@
+                . $field . ' was not setup when the product was created.
+				Please see the module documentation for more information';
+        }
+    }
+
+    if (!in_array(strtolower($params['customfields']['Plan']), $allowed_plans)) {
+        return 'JETPACK MODULE: The module does not appear to be setup correctly. ' .
+            $params['customfields']['Plan'] . ' is not an allowed plan';
+    }
+
+    if (!isset($params['configoption1']) || !isset($params['configoption2'])) {
+        return'JETPACK MODULE: Your credentials for provisioning are not complete. Please see the module documentation
+        for more information';
+    }
+    return true;
+}
+
+/**
+ * If we are attempting to get an access token and this fails parse the response
+ * http status code and response body and return a useful error from these
+ * regarding what went wrong.
+ *
+ * Return a general error if no other information is available.
+ *
+ * @param object $response Response from request for access token
+ * @return string an error string describing the issue when requesting an access token
+ */
+function get_authentication_errors_from_response($response)
+{
+    if ($response->http_status == 400 and $response->error_description) {
+        return 'JETPACK MODULE: There was a problem getting an access token for your Jetpack hosting partner 
+            account. This usually means the Client Id or Client Secret provided when setting up the module are invalid.
+            The specific error from the request was ' . $response->error_description;
+    } elseif ($response->http_status >= 500) {
+        return 'JETPACK MODULE: There was an error communicating with the provisioning server. Please try again later.';
+    }
+    return 'JETPACK MODULE: There was an error getting an authentication token. Please contact us for assistance';
+}
+
+/**
+ * If provisioning fails for a Jetpack plan parse the response http status code
+ * and response body and return a useful error regarding what went wrong. Include
+ * the response message if there is one.
+ *
+ * If the response is a 400 or 403 and there is no message in the response return
+ * a generic error letting the partner know.
+ *
+ * @param object $response Response from the provisioning request
+ * @return string an error string provided to the partner host describing the issue.
+ */
+function get_provisioning_errors_from_response($response)
+{
+    if (($response->http_status == 400 || $response->http_status == 403) && $response->message) {
+        return 'JETPACK MODULE: The following error was returned trying to provision a plan - ' . $response->message;
+    } elseif ($response->http_status >= 500) {
+        return 'JETPACK MODULE: There was an error communicating with the provisioning server. Please try again later.';
+    }
+    return 'JETPACK MODULE: There was an error provisioning the Jetpack plan. Please contact us for assistance.';
```

Comment:
Interesting. Can you expand on which cases the error message is in `$response->message` and in which case it's in `$response->error_message`?

Reply:
@ebinnion after testing this again I've actually only ever gotten $response->message and never $response->error_message testing all the scenarios I could in the partner docs for provisioning. I did go ahead and update the cancellation function to also include the $response->message if there is one. Is it possible that the partner docs need to be updated to ```message``` instead of ```error_message``` for provisioning and cancelling endpoints or are there other use cases where it may be in the response?